Models/IonAuthModel.php

```diff
@@ -374,23 +374,23 @@ public function getUserByActivationCode(string $userCode)
 	 * Validates and removes activation code.
 	 *
 	 * @param integer|string $id   The user identifier
-	 * @param boolean        $code The *user* activation code
+	 * @param string         $code The *user* activation code
 	 *                             if omitted, simply activate the user without check
 	 *
 	 * @return boolean
 	 * @author Mathew
 	 */
-	public function activate($id, bool $code=false): bool
+	public function activate($id, ?string $code=null): bool
 	{
 		$this->triggerEvents('pre_activate');
 
-		if ($code !== false)
+		if ($code)
 		{
 			$user = $this->getUserByActivationCode($code);
 		}
 		// Activate if no code is given
 		// Or if a user was found with this code, and that it matches the id
-		if ($code === false || ($user && $user->id === $id))
+		if (is_null($code) || ($user && $user->id == $id))
```

Comment:
Are the id types different? Could we instead require an int as a typehint and then typecast the db?

Reply:
The $this->ionAuth->user()->row()->id returns a string.
So you could set a string as typehint for the $id, but I don't know what the advantage of it is against the ==.
And I don't know if this datatype is only a local problem with my windows or php version, because in the database the datatype of the id is an int.

Does anyone knows more about that?

Reply:
Makes sense that it might change due to different DBs.  ID's could be strings too though if GUIDs or hashses are used.  So this seems fine to me.

Reply:
@bvrignaud what are your thoughts? I'd prefer strict comparisons and typehints, maybe we should typecast to strings?

Reply:
Have strict comparisons any security benefits or why do you prefer it ?
BTW the deactivate function has the same problem with the datatypes.
And the login function use this code to prove if a user is active:
if ($user->active === 0) So inactive user can login too.
And maybe other functions have this problem too.

Reply:
Prefer it so the types are explicit and clear.  I see your point though that this is probably an issue in quite a few places

Reply:
What db engine are you running? MySQL/Postgres/etc

Reply:
MySQL with XAMPP and php version 7.4.4

Reply:
@bvrignaud have you seen this or have any thoughts here?  I lean towards keep the explicit checks where possible and manually typecasting as needed so we can keep more explicit types throughout

Reply:
I agree.
I will do some small tests to answer you.

Reply:
As I looked.
Actually the id is a int in database, and we havn't uuid.
So if dev change the users.id for an uuid, maybe we have to leave the id (in librarie/model) without "type hints" or force it to string.